services/gossip/adapter/testkit/tampering_transport_test.go

```diff
@@ -35,15 +35,21 @@ func newTamperingHarness(logger log.Logger, ctx context.Context) *tamperingHarne
 	genesisValidatorNodes[senderAddress] = config.NewHardCodedValidatorNode(primitives.NodeAddress(senderAddress))
 	genesisValidatorNodes[listenerAddress] = config.NewHardCodedValidatorNode(primitives.NodeAddress(listenerAddress))
 
-	transport := NewTamperingTransport(logger, memory.NewTransport(ctx, logger, genesisValidatorNodes))
+	memoryTransport := memory.NewTransport(ctx, parent.Logger, genesisValidatorNodes)
+	transport := NewTamperingTransport(parent.Logger, memoryTransport)
 
 	transport.RegisterListener(listener, primitives.NodeAddress(listenerAddress))
 
-	return &tamperingHarness{
-		senderKey: senderAddress,
-		transport: transport,
-		listener:  listener,
+	harness := &tamperingHarness{
+		ConcurrencyHarness: parent,
+		senderKey:          senderAddress,
+		transport:          transport,
+		listener:           listener,
 	}
+
+	harness.Supervise(memoryTransport)
```

Comment:
Can we do `harness.Supervise(transport)` instead? (and make TamperingTransport a tree supervisor, supervising the nested transport)
`Transport` is a `ShutdownWaiter` so I think it should be simple.
May be a bit cleaner, what do you think?

Reply:
Come to think of it, `TamperingTransport` uses its own go-routines occasionally, so maybe it requires supervision anyway.